extension/icu/icu-datepart.cpp

```diff
@@ -318,9 +377,16 @@ struct ICUDatePart : public ICUDateFunc {
 					auto &child_entry = child_entries[col];
 					if (is_finite) {
 						ConstantVector::SetNull(*child_entry, false);
-						auto pdata = ConstantVector::GetData<int64_t>(*child_entry);
-						auto adapter = info.adapters[col];
-						pdata[0] = adapter(calendar, micros);
+						const auto part_index = size_t(info.part_codes[col]);
```

Comment:
Slight nit:
I think it would be nicer to turn this into a switch case with everything explicitly listed, so we can throw a NotImplemented or an Internal Exception when something gets added later that isn't covered.

Especially because the enum entries aren't explicitly numbered, it's easy to create a silent hard to debug problem when a new entry is added.
```c++
switch (part) {
	DatePartSpecifier::YEAR:
	DatePartSpecifier::MONTH:
	DatePartSpecifier::DAY: {
		auto pdata = ConstantVector::GetData<int64_t>(*child_entry);
		auto adapter = info.bigints[col];
		pdata[0] = adapter(calendar, micros);
		break;
	}
	DatePartSpecifier::JULIAN_DAY: {
		auto pdata = ConstantVector::GetData<double>(*child_entry);
		auto adapter = info.doubles[col];
		pdata[0] = adapter(calendar, micros);
		break;
	}
	default:
		throw NotImplementedException("Part not implemented for DatePartSpecifier");
}
```

Reply:
Hmm actually, looking further ahead, the functions called already do this explicit check, so it might not be necessary